client/src/components/Map/Map.js

```diff
@@ -11,20 +12,36 @@ class Map extends Component {
     },
     zoom: 16
   };
- 
+  state = {
+    lat: "",
+    lng: "",
+    center: "",
+  }
+  componentWillMount = () =>{
+    navigator.geolocation.getCurrentPosition(this.currentCoords)
+  }
+
+  currentCoords = (position) => {
+    const latitude = position.coords.latitude
+    const longitude = position.coords.longitude
+    this.setState({
+      center: {lat: latitude, lng: longitude}
+    })
+  }
+
   render() {
     return (
       // Important! Always set the container height explicitly
       <div style={{ height: '100vh', width: '100%' }}>
         <GoogleMapReact
           bootstrapURLKeys={{ key: process.env.REACT_APP_GOOGLE_API_KEY }}
           defaultCenter={this.props.center}
+          center={this.state.center}
           defaultZoom={this.props.zoom}
         >
-          <AnyReactComponent
-            lat={49.895138}
-            lng={-97.138374}
-            text="My Marker"
+          <CurrLocation
+            lat={this.state.center.lat}
```

Comment:
If the user doesn't allow location, the marker will be placed always in the center of the map (even if maps is dragged). Are we going to still draw a marker in this situation?

Reply:
I will make it so the marker only appears if the user allows his locations.